docs/spec/concepts.rst

```diff
@@ -3,97 +3,174 @@
+be subject to static checking.  The system should not signal a static type
+error for use of an expression with type :ref:`Any`.  Instead, the expression's
+value will be dynamically checked, according to the Python runtime's usual
+checks on the types of runtime values.
+
+This specification describes a gradual type system for Python.
+
+Static, dynamic, and gradual types
+----------------------------------
+
+We will refer to types that do not contain :ref:`Any` as a sub-part as **static
+types**. The dynamic type :ref:`Any` is the **dynamic type**. Any type other
+than :ref:`Any` that does contain :ref:`Any` as a sub-part is a **gradual
+type**.
+
+Static types
+~~~~~~~~~~~~
+
+A static type can intuitively be understood as denoting a set of runtime
+values. For instance, the Python static type ``object`` is the set of all
+Python objects. The Python static type ``bool`` is the set of values ``{ True,
+False }``. The Python static type ``str`` is the set of all Python strings;
+more precisely, the set of all Python objects whose runtime type (i.e.
+``__class__`` attribute) is either ``str`` or a class that inherits ``str``,
+including transitively; i.e. a type with ``str`` in its method resolution
+order. Static types can also be specified in other ways. For example,
+:ref:`Protocols` specify a static type which is the set of all objects which
+share a certain set of attributes and/or methods.
+
+The dynamic type
+~~~~~~~~~~~~~~~~
+
+The dynamic type :ref:`Any` represents an unknown static type. It denotes some
+unknown set of values.
+
+At first glance, this may appear similar to the static type ``object``, which
+represents the set of all Python objects. But it is quite different.
+
+If a value has the static type ``object``, a static type checker should ensure
+that all operations on the value are valid for all Python objects, or else emit
+a static type error. This allows very few operations! For example, if ``x`` is
+typed as ``object``, ``x.foo`` should be a static type error, because not all
+Python objects have an attribute ``foo``.
+
+A value typed as :ref:`Any`, on the other hand, should be assumed to have
+_some_ specific static type, but _which_ static type is not known. A static
+checker should not emit any errors that depend on assuming a particular static
+type; a static checker should instead assume that the runtime is responsible
+for checking the type of operations on this value, as in a dynamically-typed
+language.
+
+The subtype relation
+--------------------
+
+A static type ``B`` is a **subtype** of another static type ``A`` if (and only
+if) the set of values represented by ``B`` is a subset of the set of values
+represented by ``A``. Because the subset relation on sets is transitive and
+reflexive, the subtype relation is also transitive (if ``A`` is a subtype of
+``B`` and ``B`` is a subtype of ``C``, then ``A`` is a subtype of ``C``) and
+reflexive (``A`` is always a subtype of ``A``).
```

Comment:
There's an annoying caveat about `int` and `float` in Python.  They have a subtype relation by rule as an exception to semantic subtyping.  (Or else we repair that by saying that the set of values denoted by `float` includes all the `int` values.)

I don't know if you can do footnotes or end notes in this doc.  It seems kind of intrusive to mention this special technical case here.

Reply:
Yes, I would rather handle this case in a later section of the spec specific to ints and floats, I don't think it needs to be mentioned here. (It may already be described somewhere in the spec, I haven't yet looked.) We may need to have some discussion specifically about the intended meaning of this special case. (For instance, one possible interpretation is that the annotation `float` is in fact understood to mean `float | int`, as a special case in the annotation syntax.)

Reply:
There are also other ways to handle literal types which do not require `int <: float`.